bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/Client.scala

```diff
@@ -87,12 +91,16 @@ trait Client extends BitcoinSLogger {
     *         This future times out after 60 seconds if the client
     *         cannot be started
     */
-  def start(): Future[Unit] = {
+  override def start(): Future[BitcoindRpcClient] = {
     if (version != BitcoindVersion.Unknown) {
       val foundVersion = instance.getVersion
-      if (foundVersion != version) {
-        throw new RuntimeException(
-          s"Wrong version for bitcoind RPC client! Expected $version, got $foundVersion")
+      foundVersion match {
+        case foundVersion == V16 => this.isInstanceOf[BitcoindV16RpcClient]
```

Comment:
2 things:

1. You are using `.isInstanceOf[]` rather than `.asInstanceOf[]`. You want to use the latter 
2. Since the type signature of `start()` returns a `Future[BitcoindRpcClient]` we can just do 

```scala
this.asInstanceOf[BitcoindRpcClient]
```

Reply:
Actually looking at this in a wider context, is there anything done with this at all? Seems like we do this check and don't save it in a `val` anywhere